reana_client/cli/test.py

```diff
@@ -0,0 +1,144 @@
+
+
+@click.group(help="Workflow run test commands")
+def test_group():
+    """Workflow run test commands."""
+
+
+@test_group.command("test")
+@click.option(
+    "-n",
+    "--test-files",
+    multiple=True,
+    default=None,
+    help="Gherkin file for testing properties of a workflow execution. Overrides files in reana.yaml if provided.",
+)
+@click.pass_context
+@add_access_token_options
+@check_connection
+@add_workflow_option
+def test(ctx, workflow, test_files, access_token):
+    r"""
+    Test workflow execution, based on a given Gherkin file.
+
+    Gherkin files can be specified in the reana specification file (reana.yaml),
+    or by using the ``-n`` option.
+
+    The ``test`` command allows for testing of a workflow execution,
+    by assessing whether it meets certain properties specified in a
+    chosen gherkin file.
+
+    Example:
+        $ reana-client test -w myanalysis -n test_analysis.feature
+        $ reana-client test -w myanalysis
+        $ reana-client test -w myanalysis -n test1.feature -n test2.feature
+    """
+    try:
+        workflow_status = get_workflow_status(
+            workflow=workflow, access_token=access_token
+        )
+        status = workflow_status["status"]
+        id = workflow_status["id"]
+    except Exception:
+        display_message(f"Could not find workflow ``{workflow}``.", msg_type="error")
+        sys.exit(1)
+
+    if status != "finished":
+        display_message(
+            f"``{workflow}`` is {status}. It must be finished to run tests.",
+            msg_type="error",
+        )
+        sys.exit(1)
+
+    if not test_files:
+        reana_specification = get_workflow_specification(workflow, access_token)
+        try:
+            test_files = reana_specification["specification"]["tests"]["files"]
+        except KeyError:
+            display_message(
+                "No test files specified in reana.yaml and no -n option provided.",
+                msg_type="error",
+            )
+            sys.exit(1)
+
+    data_fetcher = DataFetcherClient(access_token)
+    for test_file in test_files:
+        click.secho(f"\nUsing test file {test_file}", fg="cyan", bold=True)
+        try:
+            results = parse_and_run_tests(id, test_file, workflow, data_fetcher)
+        except FileNotFoundError:
+            display_message(f"Test file {test_file} not found.", msg_type="error")
+            sys.exit(1)
+        except FeatureFileError as e:
+            display_message(
+                f"Error parsing feature file {test_file}: {e}", msg_type="error"
+            )
+            sys.exit(1)
+
+        click.secho(f"Summary of {test_file}:", bold=True)
+        for scenario in results[1]:
+            click.echo(f"Tested {scenario.scenario}: ", nl=False)
+            click.secho(
+                f"{scenario.result.name}",
+                fg=("green" if scenario.result == AnalysisTestStatus.passed else "red"),
+            )
```

Comment:
One last thing:  we should exit with non-zero status code (e.g. `sys.exit(1)`) if any of the tests failed